src/nsrt_learning.py

```diff
@@ -118,19 +140,26 @@ def learn_strips_operators(segments: Sequence[Segment], verbose: bool = True,
         else:
             # Get new lifted effects
             objects = {o for atom in segment.add_effects |
-                       segment.delete_effects for o in atom.objects}
+                       segment.delete_effects for o in atom.objects} | \
+                      set(segment_option_objs)
             objects_lst = sorted(objects)
             variables = [Variable(f"?x{i}", o.type)
                          for i, o in enumerate(objects_lst)]
             sub = dict(zip(objects_lst, variables))
             params.append(variables)
+            option_vars.append(tuple(sub[o] for o in segment_option_objs))
             add_effects.append({atom.lift(sub) for atom
                                 in segment.add_effects})
             delete_effects.append({atom.lift(sub) for atom
                                    in segment.delete_effects})
             new_partition = Partition([(segment, sub)])
             partitions.append(new_partition)
 
+    # We don't need option vars anymore; we'll recover them later.
+    # The only reason to include them is to make sure that params
+    # include the option vars when options are available.
```

Comment:
To check my understanding, this will happen in learn_option_specs()? If so, we can mention that in the comment.